src/driver.ts

```diff
@@ -221,34 +224,49 @@ export class AppiumFlutterDriver extends BaseDriver<FlutterDriverConstraints> {
       ...JSON.parse(JSON.stringify(args)),
     );
 
-    // HACK for eliminatin socket hang up by waiting 1 sec
     await sleep(1000);
-    // @ts-ignore
-    console.log('PageSource', await this.proxydriver.getPageSource());
+    const { appPackage, bundleId } = this.proxydriver.opts as any;
+    const packageName = appPackage || bundleId;
+    const callbacks =
+      this.proxydriver instanceof AndroidUiautomator2Driver
+        ? {
+            portForwardCallback: androidPortForward,
+            portReleaseCallback: androidRemovePortForward,
+          }
+        : this.proxydriver.isRealDevice()
+          ? {
+              portForwardCallback: iosPortForward,
+              portReleaseCallback: iosRemovePortForward,
+            }
+          : {};
+    const systemPort =
+      this.proxydriver instanceof XCUITestDriver &&
+      !this.proxydriver.isRealDevice()
+        ? null
+        : await getFreePort();
+
+    const udid = this.proxydriver.opts.udid!;
+
+    const flutterPort = await fetchFlutterServerPort({
+      udid,
+      packageName,
+      ...callbacks,
+      systemPort,
+    });
+
+    if (flutterPort == null) {
```

Comment:
please avoid using `==`. Always use strict comparison `===` where possible to avoid unexpected type conversion issues. For this particular case `if (! flutterPort)` should suffice